packages/engine/src/scene/components/MeshBVHComponent.ts

```diff
@@ -104,7 +104,7 @@ export const MeshBVHComponent = defineComponent({
                 if (toGenerate == 0) {
                   component.generated.set(true)
                 }
-                if (model.cameraOcclusion) {
+                if (model.cameraOcclusion && mesh?.visible) {
```

Comment:
weird. There is already a visible.value check above. Investigating

Reply:
this should be done at runtime anyway, or it might get desynced

maybe we just need to add VisibleComponent to the query for the camera intersection?

Reply:
Alright I went ahead and did that with a fancy filter since hits usually had a couple values